cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -0,0 +1,595 @@
+				return
+			}
+
+			deploymentDetails := TerraformDeploymentDetails{
+				ParameterFilePath: t.parametersFilePath(),
+				PlanFilePath:      t.planFilePath(),
+			}
+			if !isRemoteBackendConfig {
+				deploymentDetails.localStateFilePath = t.localStateFilePath()
+			}
+
+			result := DeploymentPlan{
+				Deployment: *deployment,
+				Details:    deploymentDetails,
+			}
+
+			asyncContext.SetResult(&result)
+		})
+}
+
+// Deploy the infrastructure within the specified template through terraform apply
+func (t *TerraformProvider) Deploy(ctx context.Context, deployment *DeploymentPlan, scope infra.Scope) *async.InteractiveTaskWithProgress[*DeployResult, *DeployProgress] {
+	return async.RunInteractiveTaskWithProgress(
+		func(asyncContext *async.InteractiveTaskContextWithProgress[*DeployResult, *DeployProgress]) {
+			t.console.Message(ctx, "Locating plan file...")
+			currentSubscription, err := t.ensureEnvSubscription(ctx)
+			if err != nil {
+				asyncContext.SetError(fmt.Errorf("failed to set az subscription , err: %w", err))
+				return
+			}
+			if currentSubscription != "" {
+				defer t.setAZSubscription(ctx, currentSubscription)
+			}
+
+			modulePath := t.modulePath()
+			terraformDeploymentData := deployment.Details.(TerraformDeploymentDetails)
+			isRemoteBackendConfig, err := t.isRemoteBackendConfig()
+			if err != nil {
+				asyncContext.SetError(fmt.Errorf("reading backend config: %w", err))
+				return
+			}
+
+			t.console.Message(ctx, "Deploying terraform template...")
+			err = asyncContext.Interact(func() error {
+				applyArgs, err := t.createApplyArgs(isRemoteBackendConfig, terraformDeploymentData)
+				if err != nil {
+					return err
+				}
+
+				runResult, err := t.cli.Apply(ctx, modulePath, applyArgs...)
+				if err != nil {
+					return fmt.Errorf("template Deploy failed: %s , err:%w", runResult, err)
+				}
+
+				return nil
+			})
+
+			if err != nil {
+				asyncContext.SetError(err)
+				return
+			}
+
+			// Set the deployment result
+			outputs, err := t.createOutputParameters(ctx, modulePath, isRemoteBackendConfig)
+			if err != nil {
+				asyncContext.SetError(fmt.Errorf("create terraform template failed: %w", err))
+				return
+			}
+
+			currentDeployment := deployment.Deployment
+			currentDeployment.Outputs = outputs
+			result := &DeployResult{
+				Deployment: &currentDeployment,
+			}
+
+			asyncContext.SetResult(result)
+		})
+}
+
+// Destroys the specified deployment through terraform destroy
+func (t *TerraformProvider) Destroy(ctx context.Context, deployment *Deployment, options DestroyOptions) *async.InteractiveTaskWithProgress[*DestroyResult, *DestroyProgress] {
+	return async.RunInteractiveTaskWithProgress(
+		func(asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress]) {
+			// discuss : check if you want to set auto-destroy . CAN WE PASS VALUES BACK AND FORTH BETWEEN THE AZD CONTEXT AND PROCESS
```

Comment:
Reading the code - it seems like the experience today is if you pass `--force` to azd then terraform won't prompt you.  Otherwise, terraform will prompt you to confirm your intention?

If so, that feels like reasonable behavior to me.

If there's still a discussion to be had, could we open an issue to track?

Reply:
Correct, now that the interactive console support has been added we can safely remove this comment.  Passing `--force` in azd will set the `-auto-approve` param in terraform.